File: AtCoder/abc/abc192/old/d.cpp
#include <iostream>
#include <string>
#include <algorithm>
using namespace std;
using ll = long long;

ll basen(string x, int d){
    ll ret = 0;
    ll base = 1;
    int n = x.size();

    for(int i = n-1; i >= 0; --i){
        ret += (x[i]-'0') * base;
        base *= d;
        if(base < 0) return -1;
    }

    return ret;
}


int main(){
    cin.tie(nullptr);
    ios::sync_with_stdio(false);

    string x;
    ll m;
    cin >> x >> m;

    int d = 0;
    for(auto c: x){
        d = max(d, c-'0');
    }

    int res = 0;
    for(int i = d+1;; ++i){
        // cout << i << " " << basen(x, i) << endl;
        ll b = basen(x, i);
        if(b!=-1 && b<=m) ++res;
        else break;
    }

    cout << res << endl;

    return 0;
}

File: AtCoder/abc/abc192/old/d.py
def basen(x, d):
    ret = 0
    base = 1
    n = len(x)

    for i in reversed(range(n)):
        if(ret > m or base > m): return False
        ret += int(x[i]) * base
        base *= d

    return ret

x = input()
m = int(input())

d = 0
for c in x:
    d = max(d, int(c))

# print(d, basen(x, d+1))
if(not basen(x, d+1) or basen(x, d+1)>m):
    print(0)
    exit()

if(len(x) == 1):
    print(1)
    exit()

ok = d+1
ng = 1<<61
while ng-ok > 1:
    mid = (ok + ng) // 2
    val = basen(x, mid)
    # print(ok, mid, ng, val)

    if(not val or val>m): ng = mid
    else: ok = mid

print(ok-d)
